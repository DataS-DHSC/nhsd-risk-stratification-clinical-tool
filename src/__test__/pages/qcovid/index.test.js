import React from 'react';
import { render } from '@testing-library/react';
import IndexPage from '../../../pages/ras/qcovid';
import QCovidContextProvider from '../../../context/qcovid/qcovid-context';

describe('QCovid Index', () => {
  it('should match snapshot ', () => {
    const props = {
      prevData: {
        csrfToken: 'csrfToken',
        headers: '{"correlationId":"5383302b-f729-4076-97d5-c428b26ef5dd"}',
      },
    };

    const container = render(
      <QCovidContextProvider>
        <IndexPage {...props} />
      </QCovidContextProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
