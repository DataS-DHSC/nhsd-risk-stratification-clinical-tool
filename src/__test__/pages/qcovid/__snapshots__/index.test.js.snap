// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QCovid Index should match snapshot  1`] = `
<DocumentFragment>
  <div
    class="nhsuk-grid-row"
  >
    <div
      class="nhsuk-grid-column-two-thirds"
    >
      <h1
        id="start-page-main-heading"
      >
        Get a COVID-19 risk assessment for a patient
      </h1>
      <form
        action="/ras/qcovid/patient-details"
        id="start-form"
        method="POST"
      >
        <input
          id="hidden-input-csrfToken"
          name="csrfToken"
          type="hidden"
          value="csrfToken"
        />
        <input
          id="hidden-input-headers"
          name="headers"
          type="hidden"
          value="{\\"correlationId\\":\\"5383302b-f729-4076-97d5-c428b26ef5dd\\"}"
        />
        <p
          id="service-description-paragraph"
        >
          Use this tool to find out a patient’s combined risk of catching and dying from coronavirus (COVID-19), and catching and subsequent hospitalisation from COVID-19.
        </p>
        <p
          id="service-model-paragraph"
        >
          The tool is based on the QCovid® risk prediction model.
        </p>
        <p
          id="service-use-paragraph"
        >
          You’ll enter information about the patient including their medical history. It should take around 3 minutes to complete. The tool has been designed for use by clinicians during a consultation with a patient or as an aid to support direct care.
        </p>
        <p
          id="risk-basis-paragraph"
        >
          The result will help you inform clinical discussions with patients about the risks if they catch COVID-19 and how they can balance those risks with their other needs. You can use the online tool to review additions to the SPL made as part of the COVID-19 Population Risk Assessment.
        </p>
        <p
          id="population-risk-assessment-link"
        >
          <a
            href="https://digital.nhs.uk/coronavirus/risk-assessment/population"
          >
            Read more about the COVID-19 Population Risk Assessment
          </a>
        </p>
        <div
          class="nhsuk-warning-callout"
          id="service-limitations"
        >
          <h3
            class="nhsuk-warning-callout__label"
          >
            Current limitations of the tool
          </h3>
          <p>
            It is based on data collected during the first peak of the pandemic. This means it cannot give an accurate result for people who are:
          </p>
          <ul
            id="service-limitations-list"
          >
            <li>
              under 19 or over 100 years old
            </li>
            <li>
              intersex or trans
            </li>
            <li>
              pregnant
            </li>
          </ul>
          <p>
            You can use the tool to estimate risk for people who were advised to shield during the first wave of the pandemic, or who have a condition that qualified a person for shielding during that time, as it may inform useful discussions with patients.
          </p>
          <p>
            However, risk may be underestimated in this group because, when the research was being carried out, these people may have been advised to shield. This means they may have been protected from catching coronavirus and therefore severe outcomes such as hospitalisation and death.
          </p>
          <p>
            There’s more detail on the 
            <a
              href="https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians#current-limitations-of-the-online-tool"
            >
              limitations of the tool in the clinical guidance
            </a>
            .
          </p>
        </div>
        <p>
          The tool uses QCovid®, which is a 'living' risk prediction model. This means that the model will be updated periodically when new research becomes available.
        </p>
        <button
          aria-disabled="false"
          class="nhsuk-button nhsuk-u-margin-bottom-6"
          id="start-now-button"
          type="submit"
        >
          Start now
        </button>
        <h2
          class="nhsuk-u-margin-top-4"
        >
          About the tool
        </h2>
        <p>
          You can read more about it, such as the:
        </p>
        <ul>
          <li>
            <a
              href="https://digital.nhs.uk/coronavirus/risk-assessment/clinical-tool/guidance-for-clinicians"
            >
              clinical guidance for the tool
            </a>
          </li>
          <li>
            <a
              href="https://www.bmj.com/content/371/bmj.m3731"
              rel="noreferrer"
            >
              British Medical Journal’s publication on the QCovid® model developed by Oxford University
            </a>
          </li>
        </ul>
        <div
          class="nhsuk-review-date"
          id="review-date"
        >
          <p
            class="nhsuk-body-s"
          >
            Page last reviewed: 1 December 2020
            <br />
            Next review due: 23 December 2023
          </p>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
